app/code/core/Mage/Eav/Model/Entity/Collection/Abstract.php

```diff
@@ -167,9 +167,11 @@ protected function _prepareStaticFields()
     protected function _initSelect()
     {
         $this->getSelect()->from(array('e' => $this->getEntity()->getEntityTable()));
-        if ($this->getEntity()->getTypeId()) {
+
+        if ($this->getEntity()->getEntityTable() != Mage_Eav_Model_Entity::DEFAULT_ENTITY_TABLE && $this->getEntity()->getTypeId()) {
```

Comment:
I believe the logic was supposed to be flipped so that the entity_type_id filter is only applied for the default `eav_entity` table (which nobody uses)?

```suggestion

        if ($this->getEntity()->getEntityTable() == Mage_Eav_Model_Entity::DEFAULT_ENTITY_TABLE && $this->getEntity()->getTypeId()) {
```